src/pages/resultsView/download/DownloadTab.tsx

```diff
@@ -1328,29 +1249,188 @@ export default class DownloadTab extends React.Component<
+                        .filter(
+                            sample =>
+                                sample.studyId in selectedProfilesGroupByStudyId
+                        )
+                        .reduce((acc: SampleMolecularIdentifier[], sample) => {
+                            acc = acc.concat(
+                                selectedProfilesGroupByStudyId[
+                                    sample.studyId
+                                ].map(profile => {
+                                    return {
+                                        molecularProfileId:
+                                            profile.molecularProfileId,
+                                        sampleId: sample.sampleId,
+                                    } as SampleMolecularIdentifier;
+                                })
+                            );
+                            return acc;
+                        }, []);
+
+                    if (sampleIdentifiers.length) {
+                        molecularData = await client.fetchMolecularDataInMultipleMolecularProfilesUsingPOST(
+                            {
+                                projection:
+                                    REQUEST_ARG_ENUM.PROJECTION_DETAILED,
+                                molecularDataMultipleStudyFilter: {
+                                    entrezGeneIds: _.map(
+                                        genes,
+                                        (gene: Gene) => gene.entrezGeneId
+                                    ),
+                                    sampleMolecularIdentifiers: sampleIdentifiers,
+                                } as MolecularDataMultipleStudyFilter,
+                            }
+                        );
+                    }
+                }
+
+                // STEP 2: generate alteration data
+                const data = {
+                    samples: _.groupBy(
+                        molecularData,
+                        data => data.uniqueSampleKey
+                    ),
+                } as CaseAggregatedData<ExtendedAlteration>;
+
+                const alterationData = generateOtherMolecularProfileData(
+                    selectedProfiles.map(profile => profile.molecularProfileId),
+                    data
+                );
+
+                // STEP 3: generate download data
+                const downloadData = generateOtherMolecularProfileDownloadData(
+                    alterationData,
+                    samples,
+                    genes
+                );
+
+                // STEP 4: download data
+                fileDownload(
+                    this.downloadDataText(downloadData),
+                    `${profileName}.txt`
                 );
-                fileDownload(textMap[profileName], `${profileName}.txt`);
             }
         );
     }
 
     @autobind
     private handleTransposedOtherMolecularProfileDownload(profileName: string) {
```

Comment:
what's difference between this method and the one above?

Reply:
merged two methods together and moved into `DownloadUtils`